cookbooks/swift/recipes/rings.rb

```diff
@@ -62,14 +65,25 @@
     cwd "/etc/swift"
   end
   (1..num_disks).each do |i|
-    j = ((i - 1) % node['nodes']) + 1
+    n_idx = ((i - 1) % node['nodes']) + 1
     z = ((i - 1) % node['zones']) + 1
     r = ((z - 1) % node['regions']) + 1
-    execute "#{service}.builder-add-sdb#{i}" do
+    dev = "sdb#{i}"
+    ip = "127.0.0.1"
+    port = "60#{n_idx}0"
+    if node['servers_per_port'] > 0 then
+      ip = "127.0.0.#{n_idx}"
+
+      # Range ports per disk per node from 60j6 - 60j9
+      # NOTE: this only supports DISKS <= 4 * NODES
+      p = 5 + (i / Float(node['nodes'])).ceil.to_int
+      port = "60#{n_idx}#{p}"
```

Comment:
Maybe... I know at least one place (a probe test?) was doing wacky shit like examining port number % 10 or % 100 or something to map back to node number or ring type... can't remember exactly.  But that's the sort of shit any change to these port numbers has a chance to break.

But it's be easy to test if your proposed change hurts any of the probe tests.